HW 02/Tast 2.2/DiagonalMatrix.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System.Text;
+using static System.Net.Mime.MediaTypeNames;
+
+namespace Tast_2._2
+{
+    public class DiagonalMatrix
+    {
+        private int _size;
+        private int[] _array;
+
+        public DiagonalMatrix(int[] diagonal)
+        {
+            if (diagonal != null)
+            {
+                _size = diagonal.Length;
+                _array = diagonal.ToArray();
+            }
+            else
+            {
+                _size = 0;
+                _array = new int[0];
+            }
+        }
+
+        public int Size
+        {
+            get { return _size; }
+        }
+
+        public int this[int i, int j]
+        {
+            get
+            {
+                if (i < 0 || i >= _size || j < 0 || j >= _size)
+                {
+                    return 0;
+                }
+                if (i == j)
+                {
+                    return _array[i];
+                }
+                return 0;
+            }
+            set
+            {
+                if (i == j)
+                {
+                    _array[i] = value;
+                }
+            }
+        }
+
```

Comment:
Sorry, can't understand the purpose of this method )) This is not an extesnion method. See
https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/extension-methods